pypeit/spectrographs/keck_hires.py

```diff
@@ -515,38 +516,84 @@ def get_detector_par(self, det, hdu=None):
             platescale      = 0.135,
             darkcurr        = 0.0,
             saturation      = 65535.,
-            nonlinear       = 0.86,
+            nonlinear       = 0.7, # Website says 0.6, but we'll push it a bit
             mincounts       = -1e10,
             numamplifiers   = 1,
-            # TODO THese are place holders at present. Getting these numbers right requires parsing headers to determine
-            # which readout mode is being used.
             gain            = np.atleast_1d([1.]),
-            ronoise         = np.atleast_1d([3.]),
+            ronoise         = np.atleast_1d([2.8]),
             )
 
-        # Detector 2. HACK
+        # Detector 2. 
         detector_dict2 = detector_dict1.copy()
         detector_dict2.update(dict(
             det=2,
             dataext=2,
             gain=np.atleast_1d([1.]),
-            ronoise=np.atleast_1d([3.])
+            ronoise=np.atleast_1d([3.1])
         ))
 
 
-        # Detector 3,. HACK
+        # Detector 3,. 
         detector_dict3 = detector_dict1.copy()
         detector_dict3.update(dict(
             det=3,
             dataext=3,
             gain=np.atleast_1d([1.]),
-            ronoise=np.atleast_1d([3.])
+            ronoise=np.atleast_1d([3.1])
         ))
 
+        # Set gain 
```

Comment:
Now I'm confused. Rather than instantiate to bogus values and then set them again below, should we instantiate to None?

Reply:
sure